lib/nerves_init_gadget/ssh_console.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Nerves.InitGadget.SSHConsole do
+  @moduledoc """
+  SSH IEx console.
+  """
+  use GenServer
+
+  @doc """
+  Since ctrl-c is intercepted in default Nerves config,
+  We need to be able to exit the shell somehow.
+  """
+  def shell_exit, do: GenServer.call(__MODULE__, :ssh_exit)
+
+  @doc false
+  def start_link(%{ssh_console_port: nil}), do: :ignore
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, [opts], name: __MODULE__)
+  end
+
+  def init([opts]) do
+    ssh = start_ssh(opts)
+    {:ok, %{ssh: ssh, opts: opts}}
+  end
+
+  def terminate(_, %{ssh: ssh}) do
+    :ssh.stop_daemon(ssh)
+  end
+
+  def handle_call(:ssh_exit, from, %{ssh: ssh} = state) do
+    # Reply to avoid crashing the call.
+    GenServer.reply(from, :ok)
+    :ok = :ssh.stop_daemon(ssh)
+    # Sleep here so when we restart, the port is open.
+    Process.sleep(5000)
+    new_ssh = start_ssh(state.opts)
+    {:noreply, :ok, %{state | ssh: new_ssh}}
+  end
+
+  defp start_ssh(%{ssh_console_port: port}) do
+    authorized_keys =
```

Comment:
It think that a note here about how we're reusing the keys from `nerves_firmware_ssh` is warranted. This feels ugly, but I don't want to list the keys twice or add console ssh to the `nerves_firmware_ssh`.

Reply:
This was the ugliest part of this feature. I don't like it either. I'll write a quick note.